core/services/scan.go

```diff
@@ -130,14 +131,14 @@ func (s *ScanService) Ready(ctx context.Context) bool {
 	return s.cveScanner.Ready(ctx)
 }
 
-func (s *ScanService) ScanAP(ctx context.Context) error {
-	ctx, span := otel.Tracer("").Start(ctx, "ScanService.ScanAP")
+func (s *ScanService) ScanAP(mainCtx context.Context) error {
+	mainCtx, span := otel.Tracer("").Start(mainCtx, "ScanService.ScanAP")
```

Comment:
Separated the main context from the context that gets created in each iteration (container)